src/config.c

```diff
@@ -2131,7 +2131,7 @@ static int numericConfigSet(typeData data, sds value, int update, const char **e
     if (data.numeric.is_memory) {
         int memerr;
         ll = memtoll(value, &memerr);
-        if (memerr || ll < 0) {
+        if (memerr) {
```

Comment:
you'll need to rebase and find a way to resolve this, the memory config infrastructure is now unsigned.
maybe we'll need to add a flag MEMORY_PERC, and actually support a notation like `10%` rather than rely on negative number and document that?

Reply:
The problem with that is that our config system performs its parsing and validation upon update and then stores the value in some global var under the `server` struct. So once the data is stored we don't have any more metadata (like the `%` notation or flag) to know how to use it.
The only other possible option instead of using a negative value is having two config variables `maxmemory_clients` and `maxmemory_clients_p`? Probably sounds familiar from somewhere...

So three options (my preferred is bold):
* Have `maxmemory-clients` and `maxmemory-clients-p`
* **Use a negative value and a custom numericConfigSet for percentages**
* Skip the idea of using percentage for `maxmemory-clients`

Reply:
Added a flag enabling config values to optionally be a percent and store the value as a negative so we know to treat it as such. I'm not sure this is such a great idea, but it seems to be ok. Please review.